Задача: Развернитесь!

piligrim-curious opened the pull request:
Main: Тесты;
Example: класс с методом reverse().

piligrim-curious committed:
Задача: Развернитесь!
Main: Тесты;
Example: класс с методом reverse().

piligrim-curious committed:
Задача: Развернитесь!
Main: Тесты;
Example: класс с методом reverse().

* Удален комментарий

MarselYsup commented on src/main/java/school/faang/turn_around/Example.java:
```diff
@@ -0,0 +1,17 @@
+
+    public static int[] reverse(int[] array) {
+
+        int[] result = new int[array.length];
```
Можно обойтись без создания дополнительного массива. Вместо этого можно инвертировать элементы массива "на месте", экономя память. Попробуй создать два указателя на первый и последний элемент, пробегаться по массиву пока указатели не встретятся и менять местами у указателей значения в массиве - так твой код будет занимать меньше памяти и менять элементы у уже имеющегося массива

MarselYsup commented on src/main/java/school/faang/turn_around/Example.java:
```diff
@@ -0,0 +1,17 @@
+public class Example {
+
+    public static int[] reverse(int[] array) {
+
```
стоит проверять, что массив не равен null, а также в случае если массив пустой или состоит из одного элемента можно не выполнять лишнюю логику

MarselYsup reviewed (changes requested):
Реализация неплохая, но можно сделать лучше, попробуй сделать через указатели👍
За тесты респект 🔥🔥🔥
но можно в тестах также проверять крайние кейсы (1 элемент, пустой массив, null)

piligrim-curious committed:
Задача: Развернитесь!
Main: Тесты;
Example: класс с методом reverse().

Изменения:

(Main.java)
* Добавлены тесты для случаев: null, [], [1].

(Example.java)
* Добавлена проверка массива на: null, [], [1];
* Удалён дополнительный массив. (изменяется исходный);
* Изменён алгоритм реверс массива ( O(n) -> O(n/2) );
* В комментариях альтернативный вариант решения задачи.
